src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -167,9 +152,77 @@ public void Delete(string key, CookieOptions options)
         });
     }
 
+    private bool TryGetLogger([NotNullWhen(true)] out ILogger? logger)
+    {
+        if (!_retrievedLogger)
+        {
+            _retrievedLogger = true;
+            var services = _features.Get<IServiceProvidersFeature>()?.RequestServices;
+            _logger = services?.GetService<ILogger<ResponseCookies>>();
+        }
+
+        logger = _logger;
+        return logger is not null;
+    }
+
+    private static MessagesToLog GetMessagesToLog(CookieOptions options)
+    {
+        var toLog = MessagesToLog.None;
+
+        if (!options.Secure && options.SameSite == SameSiteMode.None)
+        {
+            toLog |= MessagesToLog.SameSiteNotSecure;
+        }
+
+        if (!options.Secure && options.Partitioned)
+        {
+            toLog |= MessagesToLog.PartitionedNotSecure;
+        }
+
+        if (options.Partitioned && options.SameSite != SameSiteMode.None)
+        {
+            toLog |= MessagesToLog.PartitionedNotSameSiteNone;
+        }
+
+        return toLog;
+    }
+
+    private static void LogMessages(ILogger logger, MessagesToLog messages, string cookieName)
+    {
+        if ((messages & MessagesToLog.SameSiteNotSecure) != 0)
+        {
+            Log.SameSiteCookieNotSecure(logger, cookieName);
+        }
+
+        if ((messages & MessagesToLog.PartitionedNotSecure) != 0)
+        {
+            Log.PartitionedCookieNotSecure(logger, cookieName);
+        }
+
+        if ((messages & MessagesToLog.PartitionedNotSameSiteNone) != 0)
+        {
+            Log.PartitionedCookieNotSameSiteNone(logger, cookieName);
+        }
+    }
+
+    [Flags]
+    private enum MessagesToLog
+    {
+        None,
+        SameSiteNotSecure = 1 << 0,
+        PartitionedNotSecure = 1 << 1,
+        PartitionedNotSameSiteNone = 1 << 2,
+    }
+
     private static partial class Log
     {
         [LoggerMessage(1, LogLevel.Warning, "The cookie '{name}' has set 'SameSite=None' and must also set 'Secure'.", EventName = "SameSiteNotSecure")]
         public static partial void SameSiteCookieNotSecure(ILogger logger, string name);
+
+        [LoggerMessage(2, LogLevel.Warning, "The cookie '{name}' has set 'Partitioned' and must also set 'Secure'.", EventName = "PartitionedNotSecure")]
```

Comment:
```suggestion
        [LoggerMessage(2, LogLevel.Warning, "The cookie '{name}' has set 'Partitioned' and must also set 'Secure'. This cookie will likely be rejected by the client.", EventName = "PartitionedNotSecure")]
```

We might want to update our 'SameSite=None' log to do the same.